serde/src/ser/mod.rs

```diff
@@ -71,7 +71,7 @@ pub trait Serialize {
 /// reference to that state. You do not need to do any additional checks for the correctness of the
 /// state object, as it is expected that the user will not modify it. Due to the generic nature
 /// of the `Serialize` impls, modifying the object is impossible on stable Rust.
-pub trait Serializer {
+pub trait Serializer: Sized {
```

Comment:
I'll check, but unsized serializers are mostly useless anyway. I don't think they can be used to serialize anything, but I didn't check